### DISCUSSION TITLE
Permission denied

### DISCUSSION THREAD

**Discussion by lrebosio** [actions/virtual-environments#9008] [General questions] [answered]
Hello there,

I deployed a VMSS in our Azure Tenant using a runner-image Ubuntu 22.04 built with the GenerateResourceAndImage script in the repo. I also added a Custom Script to execute the post-generation scripts.

Then I added the VMSS as ADO Agents, I see that ADO deployed the Microsoft.Azure.DevOps.Pipelines.Agent extension and after a little while I can see the agent list populated by the VMSS instances.

I then run a simple pipeline just to test the whole setup, however I encounter permission errors quite often. For example, I run a simple script with some packer command, and I receive the following error when it tries to download the plugins:

Failed getting the "github.com/hashicorp/azure" plugin:
1 errors occurred:
	* could not create plugin folder "/home/dodo/.config/packer/plugins/github.com/hashicorp/azure": mkdir /home/shady1/.config: permission denied

Do I need to use a particular user when I deploy the VMSS? I'm out of ideas, what could I check?

I also read this discussion https://github.com/actions/runner-images/discussions/8191 but since I just installed the custom script extension, I'm unable to define an execution order, the ADO extension is installed and managed by ADO itself.

Thanks in advance for your help

**Comment by erik-bershel** [comment]
Hey @lrebosio!
As I see you [had already find](https://github.com/actions/runner-images/discussions/4031) an answer. Looks that in your case there is no need to run post-gen scripts in current implementation of it. It shouldn't be an issue to switch them off in your workflow. It will be very nice if you share the results. In any case feel free to reach us with other questions.

**Reply by lrebosio** [reply]
Hey @erik-bershel ,

I'm trying to perform some operations that we commonly run on our pipeline and see if I encounter any issue. I'll update this question as soon as I've something to share!

Thanks for your reply!

**Reply by lrebosio** [reply]
After some tests I encountered some difficult issues. First, I'm aware that Microsoft-Hosted Agent runs with the 'vsts' user, the VMSS runs with the AzDevOps user, and it has been said that you won't use AzDevOps during the deploy of the VMSS as primary admin user.

I deployed the VMSS with, let's say, primary admin user "pippo".

When I try several "git" commands in my pipeline I get a lot of errors about permission denied. For example, in a "git add" step I get the following error:
`warning: unable to access '/home/pippo/.config/git/ignore': Permission denied`
that I can avoid using `sudo` which is not a big deal but, on the Microsoft-Hosted Agent I don't need to run those commands with sudo.

So, I'm asking myself and, well, you all... is my approach wrong or should I just use 'sudo' and never think about it?

**Comment by erik-bershel** [comment, marked as answer]
@lrebosio you may use sudo with all the mighty power of it to access restricted areas and execute whatever you want to, sure. Also you can change permissions for all the user folder in order to grant access for other users (vsts/azdo) and binaries executed out of your user scope. Both aren't safe, but first one option is more appropriate as I suppose.

**Reply by lrebosio** [reply]
Thanks @erik-bershel
